rfcs/text/0030/process.yml

```diff
@@ -0,0 +1,97 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+- name: process
+  reusable:
+    top_level: true
+    expected:
+      - at: process
+        as: entry_leader
+        short_override: Information about the entry session leader process.
+      - at: process
+        as: session_leader
+        short_override: Information about the session leader. In some cases this may be the same information as entry_leader.
+      - at: process
+        as: group_leader
+        short_override: Information about the process group leader. In some cases this may be the same as the top level process.
+      - at: process.parent
+        as: group_leader
+        short_override: Information about the parent's process group leader. Only pid, start and entity_id fields are set.
+      - at: process.entry_leader
+        as: parent
+        short_override: Information about the entry leader's parent process. Only pid, start and entity_id fields are set.
+      - at: process.session_leader
+        as: parent
+        short_override: Information about the session leader's parent process. Only pid, start and entity_id fields are set.
+      - at: process.entry_leader.parent
+        as: session_leader
+        short_override: Information about the parent session of the entry leader. Only pid, start and entity_id fields are set.
+      - at: process.session_leader.parent
+        as: session_leader
+        short_override: Information about the parent session of the session leader. Only pid, start and entity_id fields are set.
+      - at: process
+        as: previous
+        short_override: An array of previous execution details for the process. Only executable, args, and args_count are set.
```

Comment:
Does previous need to described as only present on fork events?

Reply:
Do you mean exec and exit events??   since the fork wouldn't have a previous exec?

Reply:
whoops! exec events is what I meant, where previous would have details of the last exec or if this exec is the first one for this process, then previous would contain details of the parent that forked it.

Reply:
How bout I add.  "Set only if a process has exec'd at least once."   Since the exit event could contain this info as well.

Reply:
@m-sample 'ping'. Afaik, this is the last unresolved comment.

Reply:
Sorry, missed this - we should be specific about which type of events will have this data:

- an anemic `previous` process (only exe, args, arg_count) is present only on exec events  

perhaps with more detail on exactly how an exec is identified in the event.* fields.

Do we want to assert that it contains *all* previous execs or just the previous one?